helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -450,22 +452,17 @@ protected Map<String, ResourceAssignment> emergencyRebalance(
             resourceMap.keySet());
 
     // Step 1: Check for permanent node down
-    AtomicBoolean allNodesActive = new AtomicBoolean(true);
-    currentBestPossibleAssignment.values().parallelStream().forEach((resourceAssignment -> {
-      resourceAssignment.getMappedPartitions().parallelStream().forEach(partition -> {
-        for (String instance : resourceAssignment.getReplicaMap(partition).keySet()) {
-          if (!activeNodes.contains(instance)) {
-            allNodesActive.set(false);
-            break;
-          }
-        }
-      });
-    }));
-
+    boolean allNodesActive = currentBestPossibleAssignment.values().stream()
```

Comment:
i am not a java expert, can you please explain why we had to remove Atomic<> ?

Reply:
Thanks Komal for the review. 
Atomic<> was required for 2 reasons as it was getting used in a parallel (async) mode and secondly, becuase in stream() we cannot update a variable directly using value assignment, as it should be implicit final (so native `boolean` wouldnt have worked).. so with AtomicBoolean we can update the property of a variable, rather than the value itself. 

Other than that there is not much difference from earlier code.

Reply:
Do we know the real root case? Although Atomic can block the parallel calls. But it should finish very fast and not generate more thread?